pkg/webservice/handlers.go

```diff
@@ -211,6 +211,14 @@ func writeHeaders(w http.ResponseWriter) {
 	w.Header().Set("Access-Control-Allow-Headers", "X-Requested-With,Content-Type,Accept,Origin")
 }
 
+func buildJSONErrorResponse(w http.ResponseWriter, detail string, code int) {
+	w.WriteHeader(code)
+	errorInfo := dao.NewYAPIError(nil, code, detail)
+	if jsonErr := json.NewEncoder(w).Encode(errorInfo); jsonErr != nil {
```

Comment:
One doubt here: I think this line will write the response to the client with a successful code, correct?
how can we make sure the client-side can get the error code? Having the error code encoded in the response body is not enough.

Reply:
No, @yangwwei. We are writing the HTTP status code as part of response headers as well. Actual response body has been generated based on the following struct:

	StatusCode  int    `json:"status_code"`
	Message     string `json:"message"`
	Description string `json:"description"`

In general, client reads the status code from response headers and takes decision accordingly. I assuming this should be the case even with our UI component.

Reply:
Oh, misread that part. Thanks for the explaination.